### PR TITLE
Delete token after usage and fix inventory

### PR DESCRIPTION
**Pull request by alvarocabanas**
- Delete auth token after usage
- Add fallback device inventory

### CODE REVIEW

File: src/client/client_test.go [hunk 1]
```diff
@@ -76,6 +76,37 @@ func Test_LogIn(t *testing.T) {
...
+			res.WriteHeader(200)
+			assert.Equal(t, "a-valid-token", req.Header.Get("X-F5-Auth-Token"))
+
+			_, err := res.Write([]byte("{\"token\":\"a-valid-token\",\"another-param\":\"another\"}"))
+			assert.NoError(t, err)
+		} else {
+			res.WriteHeader(401)
+		}
+	}))
+	defer func() { testServer.Close() }()
```
Comment (gsanchezgavier) [inline src/client/client_test.go:93]: why the func() ?

File: src/f5.go [hunk 2]
```diff
@@ -54,6 +54,7 @@ func main() {
 
 	err = client.LogIn()
 	exitOnErr(err)
+	defer exitOnErr(client.LogOut())
```
Comment (gsanchezgavier) [inline src/f5.go]: not sure about the os.Exit preventing this to be executed if publish fails, but it maybe safer if we do the logout after the collectEntities

File: src/entities/system.go [hunk 3]
```diff
@@ -45,31 +45,47 @@ func CollectSystem(integration *integration.Integration, client *client.F5Client
...
+	}
+
+	var cmDevice definition.CMDevice
+	cmDeviceEndpoint := "/mgmt/tm/cm/device"
+	errDevice := client.Request(cmDeviceEndpoint, &cmDevice)
+	if errDevice == nil {
+		return cmDevice
+	}
+
+	log.Error("Couldn't get response from API for endpoint '%s': %v", smInfoEndpoint, errSmInfo)
```
Comment (gsanchezgavier) [inline src/entities/system.go:84]: this should be move after line 76 , with perhaps a debug or warn level ?

File: src/entities/system.go [hunk 4]
```diff
@@ -45,31 +45,47 @@ func CollectSystem(integration *integration.Integration, client *client.F5Client
...
+	log.Error("Couldn't get response from API for endpoint '%s': %v", cmDeviceEndpoint, errDevice)
+	return nil
```
Comment (gsanchezgavier) [inline src/entities/system.go:86]: returning an error might increase readability of this func i think.

File: src/definition/system.go [hunk 5]
```diff
@@ -1,5 +1,12 @@
 package definition
 
+type DeviceSystemInfo interface {
```
Comment (gsanchezgavier) [inline src/definition/system.go:3]: 👍

File: src/client/client_test.go (hunk 1 above)
Comment (alvarocabanas) [reply src/client/client_test.go]: fIxed

File: src/f5.go (hunk 2 above)
Comment (alvarocabanas) [reply src/f5.go]: fixed

File: src/entities/system.go (hunk 3 above)
Comment (alvarocabanas) [reply src/entities/system.go]: fixed

File: src/entities/system.go (hunk 4 above)
Comment (alvarocabanas) [reply src/entities/system.go]: done

**Pull request merged**